### PR TITLE
[train] Storage: Update XGBoost + LightGBM callback APIs

### PR DESCRIPTION
**Pull request by krfricke**
## Why are these changes needed?

The APIs are currently still using the deprecated `tune.checkpoint_dir` context manager and `tune.report`.

This PR updates XGBoost+LightGBM to use `train.report` instead. It also consolidates the available callbacks into one.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/tune/integration/xgboost.py [hunk 1]
```diff
@@ -51,33 +64,42 @@ class TuneReportCallback(TuneCallback):
...
             evals=[(test_set, "eval")],
             verbose_eval=False,
-            callbacks=[TuneReportCallback({"loss": "eval-logloss"})])
+            callbacks=[TuneReportCheckpointCallback(
+                {"loss": "eval-logloss"}, "xgboost.mdl)])
 
     """
 
+    _checkpoint_callback_cls = None
+    _report_callbacks_cls = None
```
Comment (justinvyu) [inline python/ray/tune/integration/xgboost.py:87]: Remove these?

File: python/ray/tune/integration/lightgbm.py [hunk 2]
```diff
@@ -14,17 +23,24 @@ class TuneCallback:
...
             and it will reported under the same name to Tune. If this is a
             dict, each key will be the name reported to Tune and the respective
-            value will be the metric key reported to LightGBM. If this is None,
+            value will be the metric key reported to LightGBM.
+        filename: Filename of the checkpoint within the checkpoint
+            directory. Defaults to "checkpoint". If this is None,
             all metrics will be reported to Tune under their default names as
             obtained from LightGBM.
+        frequency: How often to save checkpoints. Per default, a
+            checkpoint is saved every five iterations.
```
Comment (justinvyu) [inline python/ray/tune/integration/lightgbm.py:43]: should we set default = 0?

Comment (justinvyu) [reply python/ray/tune/integration/lightgbm.py:43]: checkpoint_at_end will already save one at the end by default. then, extra checkpointing can be added w/ this param.

**Review by justinvyu — APPROVED** [review]
Thanks! Just some minor comments.

File: python/ray/tune/integration/lightgbm.py (hunk 2 above)
Comment (krfricke) [reply python/ray/tune/integration/lightgbm.py]: Agreed - updated to 0 per default

File: python/ray/tune/integration/xgboost.py (hunk 1 above)
Comment (krfricke) [reply python/ray/tune/integration/xgboost.py:88]: We check if these properties are set for backwards compatibility with xgboost-ray and lightgbm-ray (and also for sequencing the API update).

We could test this also with `getattr` but it makes the code that invokes them more cluttered.

**Pull request merged**